contracts/MockRule.sol

```diff
@@ -65,4 +65,49 @@ contract MockRule {
         value = _value;
         receivedPayableAmount = msg.value;
     }
+
+    /**
+     * @notice Mocks CoGateway redeem function.
+     *
+     * @param _amount Redeem amount that will be transferred form redeemer
+     *                account.
+     * @param _beneficiary The address in the origin chain where the value
+     *                     tok ens will be released.
+     * @param _facilitator Facilitator address.
+     * @param _gasPrice Gas price that redeemer is ready to pay to get the
+     *                  redemption process done.
+     * @param _gasLimit Gas limit that redeemer is ready to pay
+     * @param _nonce Nonce of the redeemer address.
+     * @param _hashLock Hash Lock provided by the facilitator.
+     *
+     * @return messageHash_ which is unique for each request.
+     */
+    function redeem(
+        uint256 _amount,
+        address _beneficiary,
+        address _facilitator,
+        uint256 _gasPrice,
+        uint256 _gasLimit,
+        uint256 _nonce,
+        bytes32 _hashLock
+    )
+        public
+        payable
+        returns (bytes32)
+    {
+        require(msg.value != uint256(0), "msg.value is 0.");
```

Comment:
Why this requirement? This is not a requirement of the genuine CoGateway.redeem function.

Reply:
Here we are mocking coGateway redeem function. It's needed for below test cases:
"Verifies execution status is false when CoGateway execution fails"
"https://github.com/OpenSTFoundation/openst-contracts/blob/b72ce7a0211ba35eb207a438dde3a358be6ca5b8/test/token_holder/redeem.js#L757" 

Will add a comment above this line for clarity.